### PR TITLE
docs: refactor contributors listing

### LINKED ISSUE #609: Use Authors@R?
In the DESCRIPTION file instead of having Author and Maintainer you could use `Authors@R`. It would allow you to list all authors. pkgdown would pick it up (whereas at the moment it only picks up the maintainer).

Cc @krlmlr

### PR DESCRIPTION
**Pull request by maelle**
Fix #609

- I converted the DESCRIPTION to more standard Authors@R syntax.
- For the authors file to appear in the pkgdown website it needs to be a Markdown file and to not have a name clash with pkgdown-generated authors.html. However it might not be great to duplicate the same content? A solution might be for the author listing to live in the central igraph website and to be linked in the pkgdown website?

TODOS

- [ ] Add more authors to the DESCRIPTION file. Who are main authors to be listed there?
- [x] Add allcontributors section. I had a git2r installation issue.
- [ ] If `AUTHORS-list.md` stays, it needs to be reformatted with hard line breaks.
- [ ] If `AUTHORS-list.md` stays, fix relative link depending on where the website is deployed.

Screenshots

image

image

image

cc @krlmlr

### CODE REVIEW

**Comment by ntamas** [comment]
Okay, so, things that I can answer here:

* Historically, the main and only author of the R interface was [Gábor Csárdi](https://github.com/gaborcsardi), so he defiinitely deserves more credit than me in the author list. At some point, the maintenance of the R package was handed over to me so that's how I ended up there, but I only did some basic maintenance work, >90% of the code is still his.
* [Vincent Traag](https://github.com/vtraag), [Szabolcs Horvát](https://github.com/szhorvat) and [Fabio Zanini](https://github.com/iosonofabio) from the igraph C core team have all contributed code to the R interface at least once so they should be credited somehow.
* The author list in `AUTHORS-list.md` was assembled by Gábor at some point by actually going through every single bit of code that was lifted into igraph's C core _or_ the R interface from any other source and listing the authors of all those bits and pieces, even if the original author of the code had no active involvement in the inclusion apart from making the source code available under a license that was compatible with igraph's license. It may have been an R policy at that point in the past, or it may still be, I'm not familiar with this topic. My point is that if the author list has to be kept, I don't think that every single person on this list has to be included in the `DESCRIPTION` filel, but maybe some of these people should. But then we need to come up with clear guidelines about who should be included and who should not, and it could open up a huge can of worms so I'm not quite sure about this.

File: DESCRIPTION [hunk 1]
```diff
@@ -44,3 +42,5 @@ Config/testthat/parallel: true
 Config/testthat/start-first: vs-es, scan, vs-operators, weakref, watts.strogatz.game
 RoxygenNote: 7.2.3
 Roxygen: list(markdown = TRUE)
+Authors@R: 
+    person("Tamás", "Nepusz", , "ntamas@gmail.com", role = c("aut", "cre"), comment = "See also AUTHORS file.")
```
Comment (maelle) [inline DESCRIPTION:46]: so sorry about the typo!!

Comment (ntamas) [reply DESCRIPTION:46]: np :)

**Comment by szhorvat** [comment]
It would be nice to also set up Zenodo archiving for releases (and start issuing releases again). There's an author list to go with that as well, and Zenodo issues a DOI.

**Comment by maelle** [comment]
> Historically, the main and only author of the R interface was [Gábor Csárdi](https://github.com/gaborcsardi), so he defiinitely deserves more credit than me in the author list. At some point, the maintenance of the R package was handed over to me so that's how I ended up there, but I only did some basic maintenance work, >90% of the code is still his.

So we could list Gábor as "aut", before you in the list? (I've only seen the use of "conceptor" once https://twitter.com/ma_salmon/status/1526467262919716864, it could have been nice, as "cre" for creator is for the current maintainer only :zany_face: )

>    [Vincent Traag](https://github.com/vtraag), [Szabolcs Horvát](https://github.com/szhorvat) and [Fabio Zanini](https://github.com/iosonofabio) from the igraph C core team have all contributed code to the R interface at least once so they should be credited somehow.

The cld2 package references the author of a C++ library as cph (copyright holder) https://cran.r-project.org/web/packages/cld2/index.html Or here you could list them as "aut", or "ctb" (the difference is that by default, in the citation of an R package, you'd list the authors not the contributors).

Going forward you could use the tidyverse guideline to list external contributors (external = not in DESCRIPTION) in the NEWS file https://style.tidyverse.org/news.html which is a way to credit people without listing them in DESCRIPTION.

>    The author list in AUTHORS-list.md was assembled by Gábor at some point by actually going through every single bit of code that was lifted into igraph's C core or the R interface from any other source and listing the authors of all those bits and pieces, even if the original author of the code had no active involvement in the inclusion apart from making the source code available under a license that was compatible with igraph's license. It may have been an R policy at that point in the past, or it may still be, I'm not familiar with this topic. My point is that if the author list has to be kept, I don't think that every single person on this list has to be included in the DESCRIPTION filel, but maybe some of these people should. But then we need to come up with clear guidelines about who should be included and who should not, and it could open up a huge can of worms so I'm not quite sure about this.

So should the AUTHOR file be mentioned at all in the DESCRIPTION? Regarding legal aspects maybe https://r-pkgs.org/license.html can help.

In summary, there are many places to list people and way to filter them, but for sure you'll need some guidelines. :sweat_smile:

**Comment by maelle** [comment]
Regarding the CITATION file, if helpful: https://ropensci.org/blog/2021/02/16/package-citation/ (when getting a Zenodo DOI there's a canonical URL that'll always point to the latest DOI if I'm not mistaken)

**Comment by maelle** [comment]
I added a few people to DESCRIPTION and added the contributors section to the README. It can be updated by running `allcontributors::add_contributors()`. https://docs.ropensci.org/allcontributors/

File: README.md [hunk 2]
```diff
@@ -44,3 +44,296 @@ Please read our
...
+All contributions to this project are gratefully acknowledged using the [`allcontributors` package](https://github.com/ropenscilabs/allcontributors) following the [all-contributors](https://allcontributors.org) specification. Contributions of any kind are welcome!
+
+<table>
+
+<tr>
+<td align="center">
+<a href="https://github.com/gaborcsardi">
+<img src="https://avatars.githubusercontent.com/u/660288?v=4" width="100px;" alt=""/>
+</a><br>
+<a href="https://github.com/maelle/rigraph/commits?author=gaborcsardi">gaborcsardi</a>
```
Comment (maelle) [inline README.md:64]: probably `allcontributors::add_contributors()` would need to be run again from upstream not my fork :-)

**Comment by ntamas** [comment]
I think that @krlmlr and @maelle should also be credited in `Authors@R`. Let's talk about this tomorrow on the morning dev meeting.

**Comment by maelle** [comment]
(but thank you!!)

**Comment by ntamas** [comment]
So, regarding the author list, the decision was as follows:

* the six authors of igraph's C core should all be credited as `aut` (author) in `Authors@R`; this includes, in no particular order, Gábor Csárdi (@gaborcsardi), me (Tamás Nepusz, @ntamas), Vincent Traag (@vtraag), Szabolcs Horvát (@szhorvat), Fabio Zanini (@iosonofabio) and Daniel Noom (@GroteGnoom).
* @krlmlr should be credited as `cre` (creator) as it seems like the maintainer of the package must be credited as `ctr`
* @maelle should be credited as `ctb` (contributor)
* The Chan Zuckerberg Initiative should be credited as `fnd` (funder)

The long authors list assembled by Gábor can stay in its separate Markdown file, but there's no need to include anyone else from that list.

**Comment by maelle** [comment]
So do we keep the authors list as Markdown file, with the references to it in DESCRIPTION and the pkgdown website? It'll only be "problematic" (read: tedious) if you need to edit it as you'll need to edit both the text and Markdown versions. Also I'd need to fix the link to the rendered AUTHORS-list based on where the website is deployed.

Shouldn't CITATION be updated too?

How the pkgdown website looks with this PR.

image

image

File: DESCRIPTION [hunk 3]
```diff
@@ -46,10 +46,14 @@ RoxygenNote: 7.2.3
...
-    person("Tamás", "Nepusz", , "ntamas@gmail.com", role = c("aut", "cre"),
+    person("Tamás", "Nepusz", , "ntamas@gmail.com", role = "aut",
            comment = "See also AUTHORS file."),
     person("Vincent", "Traag", role = "aut"),
     person("Szabolcs", "Horvát", , "szhorvat@gmail.com", role = "aut"),
-    person("Fabio", "Zanini", , "fabio.zanini@fastmail.fm", role = "aut")
+    person("Fabio", "Zanini", , "fabio.zanini@fastmail.fm", role = "aut"),
+    person("Daniel", "Noom", role = "aut"),
+    person("Kirill", "Müller", role = c("aut", "cre")),
+    person("Maëlle", "Salmon", role = "ctb", email = "kirill@cynkra.com"),
```
Comment (szhorvat) [inline DESCRIPTION:56]: The email does not match the name here.

**Comment by iosonofabio** [comment]
Could you please use my fabio.zanini _AT unsw.edu.au email address?

File: DESCRIPTION (hunk 3 above)
Comment (maelle) [reply DESCRIPTION]: oops, wrong copy paste, thanks a ton for catching!

**Comment by ntamas** [comment]
> So do we keep the authors list as Markdown file, with the references to it in DESCRIPTION and the pkgdown website?

Yes, let's keep it as Markdown, and it's unlikely that it will be edited too often.

> Shouldn't CITATION be updated too?

Hmmm I don't know; this is the only paper that was written about igraph "in general". @szhorvat recently published another paper, but that's specifically about the Mathematica interface.

For fairness to other members of the dev team, I think it would be great if we could somehow get a paper published about igraph in general in the near future; right now me and Gábor are getting most of the citations for igraph due to that old paper, even though there are quite a few people involved in the development process now. Citations don't really matter to me as I'm not active in academia any more, but it's probably not the same for @vtraag or @szhorvat who should deserve more credit for their work.

@vtraag @szhorvat any suggestions / preferences for changing the citation in the R interface or shall we leave it as is?

**Comment by szhorvat** [comment]
Perhaps we could set up archiving on Zenodo, and add an extra entry with the master DOI that Zenodo generates.

You can turn on auto-archiving of all releases on Zenodo. Then everything will be done automatically when the next release is issued. This means that we need to start issuing releases (not just tagging).

It makes sense to also have a `.zenodo.json` file like for the C core, https://github.com/igraph/igraph/blob/master/.zenodo.json

**Comment by maelle** [comment]
Could igraph be a fit for https://joss.theoj.org/?

**Comment by maelle** [comment]
oh I see https://joss.theoj.org/papers/10.21105/joss.04899 :grin:

**Comment by ntamas** [comment]
`.zenodo.json` added to the `main` branch
